app/src/main/java/org/kiwix/kiwixmobile/zim_manager/ZimManageViewModel.kt

```diff
@@ -143,6 +146,12 @@ class ZimManageViewModel @Inject constructor(
     )
   }
 
+  fun actionMode(actionMode: ActionMode?) {
+    currentPage.filter { it == 1 }.subscribe {
```

Comment:
I wanted the logic, not the actual function, in the viewmodel.

there should be a field in ZimManageViewModel
```
val libraryFragmentIsVisible = currentPage.filter{ it == 1}
```

and in ZimFileSelectFragment
```
disposable.add(viewModel.libraryFragmentIsVisible.subscribe{ actionMode?.finish() })
```

if the action mode isn't finishing the maybe actionMode is null? Or the condition is activating wrong or the page change listener isn't sending the data.

`.doOnNext{ //log some stuff}`

is generally the easiest way to get an idea of what Rx is at

Reply:
> if the action mode isn't finishing the maybe actionMode is null? Or the condition is activating wrong or the page change listener isn't sending the data.

@macgills I checked all these conditions. The page change listener was sending data and the condition waa activation correctly. But for some reason, I think the first time actionMode was being sent as null. I wasn't able to figure out why though. Just for trying, I checked the value of actionMode after converting it to a string to see result, and it was same both the times the function was being called. 

I tried the way you suggested in the above comment.It works properly, I'll update the pr with that. Thanks a lot for helping with this pr. Learned a lot :)